### PR TITLE
Fix strict mode type coercion

### PR DESCRIPTION
**Pull request by tomheller**
### <strong>Pull Request</strong>

Oh the funs with strict mode 😄 This should do the trick.

#### Type of PR
Refactor

#### Checklist

- [x] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -829,6 +829,12 @@
             }
           }
         },
+        "core-js": {
```
Comment (ffriedl89) [inline package-lock.json:832]: Do you have an idea why the package.lock was changed? I guess we can revert those right?

**Review by ffriedl89 — CHANGES_REQUESTED** [review]
Thank you so much for this, going over all of them. Sorry I had to leave our liveshare session 😞

**Review by thomaspink — APPROVED** [review]

**Review by rowa-audil — APPROVED** [review]
LGTM 🐘

**Comment by tomheller** [comment]
Rebased and updated with changes from #1912

**Review by ffriedl89 — APPROVED** [review]

**Pull request merged**